### PR TITLE
feat(phone-number-input): set country and format incomplete

### PR DESCRIPTION
**Pull request by guillerot**
## Description

Default country can be set through input "country"
Incomplete number are formatted on blur

-----

Optionally, technical or more in-depth description for reviewers.
Keep an empty line under your text, as well as the 5 lines that follow it.

-----

### CODE REVIEW

File: packages/ng/forms/phone-number-input/phone-number-input.component.ts [hunk 1]
```diff
@@ -42,7 +65,7 @@ const PREFIX_ENTRIES = getCountries().map((country) => ({
 export class PhoneNumberInputComponent implements ControlValueAccessor, Validator {
 	@Input() label: string;
 
-	#onChange?: (value: E164Number | ValidatePhoneNumberLengthResult) => void;
+	#onChange?: (value: E164Number) => void;
```
Comment (Supamiu) [inline packages/ng/forms/phone-number-input/phone-number-input.component.ts:68]: The reason why we had `ValidatePhoneNumberLengthResult` was for the internal validator to be able to tell when a number is invalid and set itself to error state without having to specify some kind of `PhoneNumberFormatValidator`

Comment (guillerot) [reply packages/ng/forms/phone-number-input/phone-number-input.component.ts:68]: I have some doubts about this.
Is it a good practice to use `registerOnChange`/`onChange` for validation ?
I thought it was for control's value only, which is here the phone number

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-3060/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/22487/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-3060/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/22509/ui-diff)

**Review by Supamiu — APPROVED** [review]

**Review by vvalentin-lucca — APPROVED** [review]

**Pull request merged**